### DISCUSSION TITLE
integration for heating and cooling

### DISCUSSION THREAD

**Discussion by grericht** [nliaudat/floor-heating-controller#17] [Q&A]
Hello,
I have a heat pump and use it to heat 4 floors, each with a heating circuit distributor with 6-8 heating circuits.
So I have installed 4 FHCs with a total of 29 actuators. Basically, most things work. So far I have used the tanh option. But now I want to have the actuators controlled directly by Home Assistant. In addition to heating, my heat pump can also cool passively. I can't find an integration that is suitable for this?

**Comment by nliaudat** [comment]
For full hassio controlled valves, you can use [Hassio-custom-climate-components template. ](https://github.com/nliaudat/floor-heating-controller/wiki/Hassio-custom-climate-components)

I do not know any hassio integration which can be used for cooling and heating. 

You can modifiy the linear or tanh functions to take the absolute value of temperature differences [abs(diff_temp)]. I mean it should works, but the preliminary need is to know the heating or cooling mode of your heat pump.

All functions are linked to the difference of temperature between target and current. Applying an abs shoud works to have a valve position as the logic is the same.
